packages/loader/container-loader/src/container.ts

```diff
@@ -2432,10 +2442,39 @@ export class Container
 		return { snapshot, versionId: version?.id };
 	}
 
+	private async fetchSnapshot(
+		specifiedVersion: string | undefined,
+	): Promise<{ snapshot?: ISnapshot | ISnapshotTree; versionId?: string }> {
+		if (
+			this.mc.config.getBoolean("Fluid.Container.FetchSnapshotInNewFormat") === true &&
+			this.service?.policies?.supportNewSnapshotFormat === true
+		) {
+			const snapshot = await this.storageAdapter.getSnapshot({
+				id: specifiedVersion ?? "",
+				treeId: specifiedVersion ?? "",
+			});
+			const version: IVersion = {
+				id: snapshot?.snapshotTree.id ?? "",
+				treeId: snapshot?.snapshotTree.id ?? "",
+			};
+			this._loadedFromVersion = version;
+
+			if (snapshot === undefined && specifiedVersion !== undefined) {
+				this.mc.logger.sendErrorEvent({
+					eventName: "getSnapshotTreeFailed",
+					id: version.id,
+				});
+			}
+			return { snapshot, versionId: version?.id };
```

Comment:
nit: you use both version.id & version?.id - the latter seems like not required as it's always defined